### ISSUE TITLE
trial date does not respect the locale

### ISSUE THREAD

**Issue by scampi** [floragunncom/search-guard#615]
- version: `6.5.1-23.2`

In a JUnit test with elasticsearch and searchguard, I got the following exception where the locale of the trial date is not correctly handled:

- the params used for the test:
```
NOTE: test params are: codec=Lucene70, sim=RandomSimilarity(queryNorm=false): {}, locale=th-TH-u-nu-thai-x-lvariant-TH, timezone=America/New_York
seed=83960D12B67E3D0C
```

- the exception:
```
date_time_parse_exception: Text '๒๕๖๒-๐๑-๐๙' could not be parsed at index 0]
		at java.time.format.DateTimeFormatter.parseResolved0(DateTimeFormatter.java:1949)
		at java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1851)
		at java.time.LocalDate.parse(LocalDate.java:400)
		at com.floragunn.searchguard.configuration.SearchGuardLicense.parseDate(SearchGuardLicense.java:294)
		at com.floragunn.searchguard.configuration.SearchGuardLicense.addDays(SearchGuardLicense.java:298)
		at com.floragunn.searchguard.configuration.SearchGuardLicense.createTrialLicense(SearchGuardLicense.java:62)
		at com.floragunn.searchguard.configuration.IndexBaseConfigurationRepository.createOrGetTrial(IndexBaseConfigurationRepository.java:528)
		at com.floragunn.searchguard.configuration.IndexBaseConfigurationRepository.getLicense(IndexBaseConfigurationRepository.java:485)
		at com.floragunn.searchguard.configuration.IndexBaseConfigurationRepository.reloadConfiguration(IndexBaseConfigurationRepository.java:331)
		at com.floragunn.searchguard.action.configupdate.TransportConfigUpdateAction.nodeOperation(TransportConfigUpdateAction.java:113)
		at com.floragunn.searchguard.action.configupdate.TransportConfigUpdateAction.nodeOperation(TransportConfigUpdateAction.java:47)
		at org.elasticsearch.action.support.nodes.TransportNodesAction.nodeOperation(TransportNodesAction.java:140)
		at org.elasticsearch.action.support.nodes.TransportNodesAction$NodeTransportHandler.messageReceived(TransportNodesAction.java:260)
		at org.elasticsearch.action.support.nodes.TransportNodesAction$NodeTransportHandler.messageReceived(TransportNodesAction.java:256)
		at com.floragunn.searchguard.ssl.transport.SearchGuardSSLRequestHandler.messageReceivedDecorate(SearchGuardSSLRequestHandler.java:170)
		at com.floragunn.searchguard.transport.SearchGuardRequestHandler.messageReceivedDecorate(SearchGuardRequestHandler.java:260)
		at com.floragunn.searchguard.ssl.transport.SearchGuardSSLRequestHandler.messageReceived(SearchGuardSSLRequestHandler.java:142)
		at com.floragunn.searchguard.SearchGuardPlugin$7$1.messageReceived(SearchGuardPlugin.java:639)
		at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:66)
		at org.elasticsearch.transport.TcpTransport$RequestHandler.doRun(TcpTransport.java:1350)
		at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:723)
		at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
		at java.lang.Thread.run(Thread.java:748)
```

It seems that the date is formatted with the locale `th-TH-u-nu-thai-x-lvariant-TH` at https://github.com/floragunncom/search-guard/blob/f6f2e69b5b007bc7e106d8601ae91d9f5029bcb2/src/main/java/com/floragunn/searchguard/configuration/IndexBaseConfigurationRepository.java#L463 but the locale is not taken into account when the date is parsed at 
https://github.com/floragunncom/search-guard/blob/f6f2e69b5b007bc7e106d8601ae91d9f5029bcb2/src/main/java/com/floragunn/searchguard/configuration/SearchGuardLicense.java#L294

**Comment by floragunncom** [comment]
I can reproduce this issue but i'am not sure how to fix it. I adjusted one of our unit tests to run with `Locale.setDefault(Locale.forLanguageTag("th-TH-u-nu-thai-x-lvariant-TH"));`.
The DEFAULT_FOMATTER has the correct locale set (`DEFAULT_FOMATTER.getLocale()` reports th_TH_TH_#u-nu-thai) so i assume LocalDate.parse should be able to parse `๒๕๖๒-๐๑-๐๙`

**Comment by floragunncom** [comment]
What do think of https://github.com/floragunncom/search-guard/pull/617 ?

**Comment by floragunncom** [comment]
fixed with v24.1
